### PR TITLE
Feature/#2 move migrations to fees

### PR DESCRIPTION
**Pull request by jacob-hyde**
1. Adds Support package
2. Adds required models to config/fees.php
3. Moves Migration to Fees package with a down method
4. Moves Fee model to Fees package
5. Moves FeeFactory to Fees package
6. Had to override the migrations from the Spatie package so we can use the date as when they were created

Removed the function from Models/Fee.php (Fee Model) which belongs in the Cart package.
```
    /**
     * Generate fees by cart item.
     *
     * @return int
     */
    public function generateTotalFeesByCartItem(CartItem $cartItem): int
    {
        $fee = 0;

        switch ($this->applies_to) {
            case self::APPLIES_TO_PRODUCT:
                break;
            case self::APPLIES_TO_BOOKING:
            case self::APPLIES_TO_EACH:
                if (! empty($this->percent)) {
                    $fee = $cartItem->amount * ($this->percent / 100);
                }
                if (! empty($this->amount)) {
                    $fee += $this->amount;
                }

                break;
            case self::APPLIES_TO_PARTICIPANT:
                $fee = $cartItem->participants * $this->amount;

                break;
        }

        return $fee;
    }
```

### CODE REVIEW

**Review by joshtorres — COMMENTED** [review]
Looks good from what I can see. Please let me know once you are able to test it working in the application.

File: config/fees.php [hunk 1]
```diff
@@ -1,5 +1,12 @@
 <?php
 
 return [
+    //Users Model
+    'user_model' => \App\Models\User::class,
```
Comment (sl0wik) [inline config/fees.php:5]: Should be updated accordingly to https://www.notion.so/tger/Rules-c1a7ceb5485048d0b2b68ed4247f887b

Comment (phuclh) [reply config/fees.php:5]: @sl0wik Updated in my latest commit.

**Review by drewroberts — APPROVED** [review]
Thank you! 🎈

**Pull request merged**